Add MPTs to xrpl.org docs (#2908)
XRPLF/xrpl-dev-portal: docs/references/protocol/transactions/types/mptokenissuancecreate.md

---
html: mptokenissuancecreate.html
parent: transaction-types.html
blurb: Issue a new Multi-purpose Token.
labels:
 - Multi-purpose Tokens, MPTs
---

# MPTokenIssuanceCreate

> [[Source]](https://github.com/XRPLF/rippled/blob/master/src/ripple/app/tx/impl/MPTokenIssuanceCreate.cpp "Source")

Reviewer: ```suggestion
[[Source]](https://github.com/XRPLF/rippled/blob/master/src/xrpld/app/tx/detail/MPTokenIssuanceCreate.cpp "Source")
```

_(Requires the [MPToken amendment][] {% not-enabled /%})_

> The `MPTokenIssuanceCreate` transaction creates an [MPTokenIssuance](../../data-types/mptokenissuance.md) object and adds it to the relevant directory node of the creator account. This transaction is the only opportunity an issuer has to specify any token fields that are defined as immutable (for example, MPT Flags).

Reviewer 2: ```suggestion
The `MPTokenIssuanceCreate` transaction creates an [MPTokenIssuance](../../ledger-data/ledger-entry-types/mptokenissuance.md) object and adds it to the relevant directory node of the creator account. This transaction is the only opportunity an issuer has to specify any token fields that are defined as immutable (for example, MPT Flags).
```

> The `MPTokenIssuanceCreate` transaction creates an MPTokenIssuance <!--[MPTokenIssuance](mptokenissuance.html)--> object and adds it to the relevant directory node of the creator account. This transaction is the only opportunity an issuer has to specify any token fields that are defined as immutable (for example, MPT Flags).

Reviewer: ```suggestion
The `MPTokenIssuanceCreate` transaction creates an [MPTokenIssuance](../../data-types/mptokenissuance.md) object and adds it to the relevant directory node of the creator account. This transaction is the only opportunity an issuer has to specify any token fields that are defined as immutable (for example, MPT Flags).
```

If the transaction is successful, the newly created token is owned by the account (the creator account) that executed the transaction.

Whenever your query returns an `MPTokenIssuance` transaction response, there will always be an `mpt_issuance_id` field on the Transaction Metadata page.

## Example MPTokenIssuanceCreate JSON

This example assumes that the issuer of the token is the signer of the transaction.

```json

> Example MPTokenIssuanceCreate transaction

Reviewer: ```suggestion
```

{
  "TransactionType": "MPTokenIssuanceCreate",
  "Account": "rajgkBmMxmz161r8bWYH7CQAFZP5bA9oSG",

>   "AssetScale": "2",

Reviewer: ```suggestion
  "AssetScale": 2,
```

  "TransferFee": 314,
  "MaximumAmount": "50000000",
  "Flags": 83659,

>   "MPTokenMetadata": "FOO",

Reviewer 3: do we have a schema for metadata? if so, should we put a valid metdata example?

>   "MPTokenMetadata": "FOO",

Reviewer: shoudn't be hex string?

>   "Fee": 10

Reviewer: ```suggestion
  "Fee": "10"
```

}
```

## MPTokenIssuanceCreate Fields

> <!-- {% include '_snippets/tx-fields-intro.md' %} -->

Reviewer: ```suggestion
{% raw-partial file="/docs/_snippets/tx-fields-intro.md" /%}
```

| Field           | JSON Type           | [Internal Type][] | Description        |
|:----------------|:--------------------|:------------------|:-------------------|

> | `TransactionType` | object              | UInt16            | Indicates the new transaction type MPTokenIssuanceCreate. |

Reviewer 3: ```suggestion
| `TransactionType` | string              | UInt16            | Indicates the new transaction type MPTokenIssuanceCreate. |
```

| `Flags`           | number              | UInt16            | Specifies the flags for this transaction. See [MPTokenIssuanceCreate Flags](#mptokenissuancecreate-flags). |
| `TransferFee`      | number             | UInt16            | (Optional) The value specifies the fee to charged by the issuer for secondary sales of the Token, if such sales are allowed. Valid values for this field are between 0 and 50,000 inclusive, allowing transfer rates of between 0.000% and 50.000% in increments of 0.001. The field _must not_ be present if the tfMPTCanTransfer flag is not set. If it is, the transaction should fail and a fee should be claimed. |
| `MaximumAmount`   | string              | UInt64            | (Optional) The maximum asset amount of this token that should ever be issued, as a base 10 encoded string. For issuances that do not have a maximum limit, set this value to 9,223,372,036,854,775,807 (2^63-1). |
| `MPTokenMetadata` | string              | Blob              | Arbitrary metadata about this issuance, in hex format. The limit for this field is 1024 bytes. |

## MPTokenIssuanceCreate Flags

> Transactions of the MPTokenIssuanceCreate type support additional values in the `Flags` field <!-- [`Flags` field](transaction-common-fields.html#flags-field)-->, as follows:

Reviewer: ```suggestion
Transactions of the MPTokenIssuanceCreate type support additional values in the [`Flags` field](../common-fields.md#flags-field), as follows:
```

_(Requires the [MPToken amendment][] {% not-enabled /%})_

The `MPTokenIssuanceCreate` transaction creates an MPTokenIssuance <!--[MPTokenIssuance](mptokenissuance.html)--> object and adds it to the relevant directory node of the creator account. This transaction is the only opportunity an issuer has to specify any token fields that are defined as immutable (for example, MPT Flags).

If the transaction is successful, the newly created token is owned by the account (the creator account) that executed the transaction.

Whenever your query returns an `MPTokenIssuance` transaction response, there will always be an `mpt_issuance_id` field on the Transaction Metadata page.

## Example MPTokenIssuanceCreate JSON

This example assumes that the issuer of the token is the signer of the transaction.

```json
Example MPTokenIssuanceCreate transaction
{
  "TransactionType": "MPTokenIssuanceCreate",
  "Account": "rajgkBmMxmz161r8bWYH7CQAFZP5bA9oSG",
  "AssetScale": "2",
  "TransferFee": 314,
  "MaximumAmount": "50000000",
  "Flags": 83659,
  "MPTokenMetadata": "FOO",
  "Fee": 10
}
```

> ## MPTokenIssuanceCreate Fields

Reviewer 2: Apply this along with suggestion from tequdev to avoid duplicate headings.
```suggestion
<!-- ## MPTokenIssuanceCreate Fields -->
```
